engineyard/ey-cookbooks-stable-v5
# [PR282] Added an ability to change puma restart sleep timeout value

## Description of your patch

Adding an ability to customize Puma restart sleep timeout value.

## Recommended Release Notes

Ability to customize Puma restart sleep timeout value added

## Estimated risk

Low

## Components involved

Puma application server

## Description of testing done

See QA instructions

## QA Instructions

1. Create a simple Ruby application.  (https://github.com/engineyard/todo.git) can be used for this testing.
2. Boot the environment with QA stack created from the PR282_improvement branch. Environment should be run with following parameters:
  * Environment type - Single Instance.
  * Application Server Stack - Puma.
3. Edit file `/etc/chef/recipes/cookbooks/puma/attributes/default.rb` change sleep_timeout value to 10.
4. Run Chef again `PATH=/usr/local/ey_resin/bin:$PATH /home/ey/bin/chef-solo -j /etc/chef/dna.json -c /etc/chef/solo.rb`.
5. Run command `cat /engineyard/bin/app_qa_testing_for_solving_cc | grep sleep` to ensure that sleep value was succesfully changed to the value provided in the attributes file.
